### PR TITLE
wxPython: 3.0.0.0 -> 3.0.2.0 & refactor to use buildPythonPackage

### PR DESCRIPTION
**Pull request by vbgl**

### CODE REVIEW

**Comment by domenkozar** [comment]
Thanks!! Can you also unite that code with `2.8`?

**Comment by vbgl** [comment]
I gave a try to the generic.nix pattern to unite the derivation for the two versions of wxPython.

The wxPython packages are now enabled only for python2.7 (there is no upstream support for python3).

Travis failed but the rebuild went through locally without error.

File: pkgs/development/python-modules/wxPython/generic.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  preConfigure = "cd wxPython";
+
+  setupPyBuildFlags = [ "WXPORT=gtk2" "NO_HEADERS=1" "BUILD_GLCANVAS=${if openglSupport then "1" else "0"}" "UNICODE=1" ];
+
+  installPhase = ''
+    ${python}/bin/${python.executable} setup.py ${stdenv.lib.concatStringsSep " " setupPyBuildFlags} install --prefix=$out
+  '';
+
+  inherit openglSupport;
```
Comment (domenkozar) [inline pkgs/development/python-modules/wxPython/generic.nix]: this is redundant if you set `passthru`

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -15023,11 +15023,13 @@ let
 
   wxPython = self.wxPython28;
 
-  wxPython28 = callPackage ../development/python-modules/wxPython/2.8.nix {
+  wxPython28 = import ../development/python-modules/wxPython/2.8.nix {
```
Comment (domenkozar) [inline pkgs/top-level/python-packages.nix]: Why did you get rid of `callPackage`? that removes ability to override derivation or the function

**Comment by vbgl** [comment]
Thanks for your comments. I have removed the redundant `openglSupport` attribute.

Concerning `callPackage`, I’ve taken inspiration from the `gnutls` package (https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L6527). The derivation still use `callPackage`, but in the files `2.8.nix` and `3.0.nix`.

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (vcunat) [inline pkgs/top-level/python-packages.nix]: It is used for the real call inside, so IMO the result should be overridable. These imports are just thin wrappers.

**Pull request merged**